### PR TITLE
Properly handle --gen-pkg-config/--gen-script with internal libraries

### PR DESCRIPTION
**Pull request by ezyang**
Just the last three commits.

### CODE REVIEW

**Comment by ezyang** [comment]
Can we kick the Travis build?

**Comment by ezyang** [comment]
Apparently, I have no clue how to execute bat files on Windows (which is why AppVeyor is failing.)

File: Cabal/Distribution/Simple/Register.hs [hunk 1]
```diff
@@ -127,28 +148,40 @@ registerOne pkg lbi regFlags lib
...
                     ++ maybeToList (flagToMaybe  (regPackageDB regFlags))
-    distPref  = fromFlag (regDistPref regFlags)
     verbosity = fromFlag (regVerbosity regFlags)
 
-    writeRegistrationFile installedPkgInfo = do
-      notice verbosity ("Creating package registration file: " ++ regFile)
-      writeUTF8File regFile (IPI.showInstalledPackageInfo installedPkgInfo)
-
-    writeRegisterScript installedPkgInfo =
+    writeRegistrationFile = do
```
Comment (23Skidoo) [inline Cabal/Distribution/Simple/Register.hs]: The name should probably reflect that it can generate a directory.

**Comment by 23Skidoo** [comment]
LGTM.

**Comment by ezyang** [comment]
Previous AppVeyor build failed for no discernible reason. Maybe it will pass this time.

**Pull request closed without merging**